libraries/base/GHC/IO/Exception.hs

```diff
@@ -231,7 +232,7 @@ data ExitCode
                 -- The exact interpretation of the code is
                 -- operating-system dependent.  In particular, some values
                 -- may be prohibited (e.g. 0 on a POSIX-compliant system).
-  deriving (Eq, Ord, Read, Show, Typeable)
+  deriving (Eq, Ord, Read, Show, Generic, Typeable)
```

Comment:
This should mostly be harmless, but it may create some inconsistencies when compiling Hackage packages. You can keep this as it is, but can you add a warning here that this may lead to some problems later on?